hal/src/rtl872x/gpio_hal.cpp

```diff
@@ -433,3 +434,64 @@ uint32_t hal_gpio_pulse_in(hal_pin_t pin, uint16_t value) {
 
     return (hal_timer_micros(nullptr) - pulse_start);
 }
+
+int hal_gpio_get_drive_strength(hal_pin_t pin, hal_gpio_drive_t* drive) {
+    uint32_t rtlPin = hal_pin_to_rtl_pin(pin);
+
+    /* get PADCTR */
+    uint32_t drvStrength = PINMUX->PADCTR[rtlPin];
+
+    /* get Pin_Num drvStrength contrl */
+    drvStrength &= PAD_BIT_MASK_DRIVING_STRENGTH << PAD_BIT_SHIFT_DRIVING_STRENGTH;
+    drvStrength >>= PAD_BIT_SHIFT_DRIVING_STRENGTH;
+
+    // Pad driving strength
+    //   - PAD_DRV_STRENGTH_0: 4mA
+    //   - PAD_DRV_STRENGTH_1: 8mA   (Severe overshoot, not recommended to use it)
+    //   - PAD_DRV_STRENGTH_2: 12mA
+    //   - PAD_DRV_STRENGTH_3: 16mA  (Severe overshoot, not recommended to use it)
+    switch(drvStrength) {
+        case PAD_DRV_STRENGTH_0: *drive = HAL_GPIO_DRIVE_STANDARD; break;
+        case PAD_DRV_STRENGTH_2: *drive = HAL_GPIO_DRIVE_HIGH; break;
+        // DVOS won't configure PAD_DRV_STRENGTH_1 and PAD_DRV_STRENGTH_3 due to the
+        // severe overshoot, so it is the default setting
+        default:
+            *drive = HAL_GPIO_DRIVE_DEFAULT; break;
+    }
+
+    return SYSTEM_ERROR_NONE;
+}
+
+int hal_gpio_configure_drive_strength(hal_pin_t pin, hal_gpio_drive_t drive) {

```

Comment:
Would it be clearer if this function was renamed `hal_gpio_set_drive_strength()` to match `hal_gpio_get_drive_strength()`?

Reply:
To be honest, I'd also prefer `hal_gpio_set_drive_strength()`, I intended to keep it consistent with `hal_gpio_configure`, well, let's rename it.